agent/src/agent/data_extractor/snmp/snmp.py

```diff
@@ -0,0 +1,83 @@
+import time
+
+from pysnmp.entity.engine import SnmpEngine
+from pysnmp.hlapi import getCmd, CommunityData, UdpTransportTarget, ContextData
+from pysnmp.smi.rfc1902 import ObjectType, ObjectIdentity
+from agent.data_extractor.snmp.delta_calculator import DeltaCalculator
+from agent.modules import logger
+from agent.pipeline import Pipeline
+from urllib.parse import urlparse
+
+logger_ = logger.get_logger(__name__, stdout=True)
+delta_calculator = DeltaCalculator()
+
+HOSTNAME_OID = '1.3.6.1.2.1.1.5.0'
+
+
+def extract_metrics(pipeline_: Pipeline) -> list:
+    url = urlparse(pipeline_.source.url)
+    iterator = getCmd(
+        SnmpEngine(),
+        CommunityData(pipeline_.source.read_community, mpModel=0),
+        UdpTransportTarget((url.hostname, url.port), timeout=pipeline_.source.query_timeout, retries=0),
+        ContextData(),
+        *[ObjectType(ObjectIdentity(mib)) for mib in pipeline_.config['mibs']],
+        lookupNames=True,
+        lookupMib=True
+    )
+
+    metrics = []
+    for response in iterator:
+        error_indication, error_status, error_index, var_binds = response
+        if error_indication:
+            logger_.error(error_indication)
+            continue
+        elif error_status:
+            logger_.error('%s at %s' % (
+                error_status.prettyPrint(),
+                error_index and var_binds[int(error_index) - 1][0] or '?'
+            ))
+            continue
```

Comment:
shouldn't we send some kind of error in the response? code 500, for example. Or send a metric to monitoring. Otherwise we won't be able to know if there were any errors we'll just get no data